ui/admin/app/components/form/target/add-credential-sources/index.js

```diff
@@ -0,0 +1,36 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { A } from '@ember/array';
+
+export default class FormTargetAddCredentialSourcesIndexComponent extends Component {
+  // =properties
+
+  /**
+   * Array of selected credential source IDs.
+   * @type {EmberArray}
+   */
+  selectedCredentialSourceIDs = A();
+
+  // =actions
+
+  /**
+   * Add/Remove credential source to current selection
+   * @param {CredentialLibraryModel, CredentialModel} credentialSource
+   */
+  @action
+  toggleCredentialSource(credentialSource) {
+    if (!this.selectedCredentialSourceIDs.includes(credentialSource.id)) {
+      this.selectedCredentialSourceIDs.addObject(credentialSource.id);
+    } else {
+      this.selectedCredentialSourceIDs.removeObject(credentialSource.id);
+    }
+  }
+
+  /**
+   * Submit selected credential source ids
+   */
+  @action
+  submit(fn) {
+    fn(this.selectedCredentialSourceIDs);
```

Comment:
Do we need to pass another function through the submit action? Could the passed in submit have been accessed with `this.args.submit`?

Or on the template use `@onSubmit={{fn @submit this.selectedCredentialSourceIDs}}`?

Not sure if there's a best practice but it feels a little weird to have to pass in another anonymous function like this, but maybe that's just me

Reply:
That is a good question. I think this might just be the existing practice. This might be one of those things were we evaluate this as a team to decide on the best practice going forward. Any thoughts on this @randallmorey?

Reply:
You can try without the action and see if it works.

Reply:
I will try this in a different branch so I am not holding up my team with this PR.